src/PIL/DdsImagePlugin.py

```diff
@@ -9,103 +9,255 @@
 Full text of the CC0 license:
   https://creativecommons.org/publicdomain/zero/1.0/
 """
-
+import io
 import struct
-from io import BytesIO
+from enum import IntEnum, IntFlag
 
 from . import Image, ImageFile
+from ._binary import i32le as i32
 from ._binary import o32le as o32
 
 # Magic ("DDS ")
 DDS_MAGIC = 0x20534444
 
-# DDS flags
-DDSD_CAPS = 0x1
```

Comment:
Please don't delete these, it's a backwards incompatible change in case someone was using `DdsImagePlugin.DDSD_CAPS` etc.

(I've not reviewed the rest of this file.)